### ISSUE TITLE
onBeforeRender called twice per render when passing a Mesh to the renderer.

### ISSUE THREAD

**Issue by sjpt** [mrdoob/three.js#19477]
##### Description of the problem
When onBeforeRender is applied to a mesh, and that mesh is rendered directly (not via a scene), then the onBeforeRender callback is applied twice for each render call.

I appreciate that it is not standard practice to render a mesh directly, but there are cases where it is helpful to avoid (the very small) overheads in performance, and in code noise.

https://jsfiddle.net/sjpt/t664p2yn/

##### Three.js version

- [x ] Dev
- [x ] r116
- [ ] ...

##### Browser

- [x] All of them
- [ ] Chrome
- [ ] Firefox
- [ ] Internet Explorer

##### OS

- [x] All of them
- [ ] Windows
- [ ] macOS
- [ ] Linux
- [ ] Android
- [ ] iOS

**Comment by Mugen87** [comment]
I think we have to change this line:
https://github.com/mrdoob/three.js/blob/452a9e93188410cbe431eb6af38506c5a6560a79/src/renderers/WebGLRenderer.js#L1174
to 
```js
if( scene.isScene ) scene.onBeforeRender( _this, scene, camera, renderTarget || _currentRenderTarget ); 
```
and do the same thing for `onAfterRender()`.

**Comment by WestLangley** [comment]
>I think we have to change this line:

The renderer expects a `Scene` as the first argument, so we do not have to do anything.

If you do not think a `Scene` as the first argument is appropriate, then the first step is to change the API.

**Comment by Mugen87** [comment]
@mrdoob recently implemented a342f06618f253d5b43aa3c9d604664b317aa034 or 4cc880d5b6ea937343b987c906ebbf2bfe066e54 so I think applying 3D objects directly to `render()` is valid.

**Comment by sjpt** [comment]
Associated comment. onBeforeRender() does not seem to get called when applied THREE.Group.

**Comment by Mugen87** [comment]
That's correct. It's only called for renderable 3D objects and scenes.

**Comment by sjpt** [comment]
That should be clearer in the documentation.

Is there any reason not to apply it to groups?
I have group Group consisting of objects Wire and Shaded.
Both Wire and Shaded require the same preparatory work Prep().
I had assumed that setting `Group.onBeforeRender=Prep` would do that.

**Comment by Mugen87** [comment]
> Is there any reason not to apply it to groups?

This was already discussed here: #11306

**Comment by sjpt** [comment]
That #11306 thread seems to say that it should work,
'mrdoob added this to the rXX milestone on 9 Mar 2018'

Came up again in #14970 which was closed in favour of #11306, but #11306 still isn't resolved.
